src/extensions/default/Microsoft.DotNet.UpgradeAssistant.Extensions.Default/DefaultExtensionServiceProvider.cs

```diff
@@ -74,7 +74,6 @@ private static void AddPackageReferenceAnalyzers(IServiceCollection services)
         {
             // Add package analyzers (note that the order matters as the analyzers are run in the order registered)
             services.AddTransient<IDependencyAnalyzer, DuplicateReferenceAnalyzer>();
-            services.AddTransient<IDependencyAnalyzer, TransitiveReferenceAnalyzer>();
```

Comment:
Instead of removing this, we could just put it last in the list and stop looping in the dependency analyzer step. I'm king of torn which approach is best, but I think just putting this analyzer last solves the problem and doesn't introduce another upgrade step, which is nice.

Reply:
I agree - but if someone adds an analyzer after, it's last. Maybe we could add a `OrderAttribute` that would allow some sorting....

Reply:
Yea, I considered that. Adding comments when we register the analyzers would almost be good enough. The challenge would be if more analyzers are added in a different extension. Hm...

I guess the options are either to have some sort of `OrderAttribute` like you suggest (a number?) or to pull it out into a separate step. Neither seems like a great solution as they both introduce some complexity. Either way is fine with me.

Reply:
Added an ordering attribute and associated changes/tests for that